src/ai/backend/client/cli/params.py

```diff
@@ -166,3 +168,26 @@ def convert(self, arg, param, ctx):
                 return arg.split(",")
         except ValueError as e:
             self.fail(repr(e), param, ctx)
+
+
+class OptionalType(click.ParamType):
+    name = "Optional Type Wrapper"
+
+    def __init__(self, type_: type) -> None:
+        super().__init__()
+        self.type_ = type_
+
+    def convert(self, value: Any, param, ctx):
+        try:
+            if value is None or value is undefined:
+                return value
+            else:
+                try:
+                    value = int(value)
+                    self.type_ = int
+                except ValueError:
+                    pass
```

Comment:
```suggestion
                return self.type_(value)
```
`OptionalType(int)` cannot convert the input `str` value into `int`?
If not, we should think of general way to handle the value rather hard-parse it into `int`

Reply:
> OptionalType(int) cannot convert the input str value into int?

Oh, it can! I have refactored it